python/plugins/grassprovider/Grass7AlgorithmProvider.py

```diff
@@ -31,11 +31,12 @@
 from processing.core.ProcessingConfig import (ProcessingConfig, Setting)
 from grassprovider.Grass7Utils import Grass7Utils
 from grassprovider.Grass7Algorithm import Grass7Algorithm
-from processing.tools.system import isWindows, isMac
+from processing.tools.system import isWindows, isMac, mkdir
 
 
 class Grass7AlgorithmProvider(QgsProcessingProvider):
-    descriptionFolder = Grass7Utils.grassDescriptionPath()
+    descriptionFolders = Grass7Utils.grassDescriptionFolders()
+    mkdir(descriptionFolders[0])
```

Comment:
this looks a little fragile -- if someone modifies the return order of grassDescriptionFolders then we'll potentially be trying to make a path in a read-only location.

Could you move the user folder logic out of grassDescriptionFolders into a standalone function userDescriptionFolder() , which is then called by grassDescriptionFolders?

Reply:
OK, I've done that, although I'm not sure it is really much less fragile.
- it errors if the userDescriptionFolder can't be created, preventing the provider from loading.  This is similar to the models provider failing to load if the default models folder can't be created (even if there is another existing folder in the ModelerUtils.MODELS_FOLDER setting).  Is this behaviour OK?
- if someone did reverse the return order they would make it impossible for users to override the built-in GRASS algorithms, which is not really what we want.